### PR TITLE
Merging what I have so far

### PR DESCRIPTION
**Pull request by MrSoup678**
This also bumps osu-deploy to upstream's master

### CODE REVIEW

File: AppDir/AppRun [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+HERE="$(dirname "$(readlink -f "${0}")")"
+export PATH="${HERE}"/usr/bin/:"${PATH}"
+EXEC=$(grep -e '^Exec=.*' "${HERE}"/*.desktop | head -n 1 | cut -d "=" -f 2 | cut -d " " -f 1)
+LD_LIBRARY_PATH=$HERE exec "${HERE}/${EXEC}" $@
```
Comment (sr229) [inline AppDir/AppRun:5]: is `LD_LIBRARY_PATH` necessary? I think we compile osu as a self-contained app.

**Review by sr229 — APPROVED** [review]
Just a little nit, you can answer this later but I'm merging this.

File: AppDir/AppRun (hunk 1 above)
Comment (MrSoup678) [reply AppDir/AppRun:5]: We do, however osu uses external lib - BASS... which is located at the same directory as the osu bin. Running osu! without LD_LIBRARY_PATH set will crash osu.

Comment (MrSoup678) [reply AppDir/AppRun:5]: See https://github.com/ppy/osu/blob/master/README.md for more info.

**Pull request merged**